### PR TITLE
feat: add aggregate actions hook

### PR DESCRIPTION
**Pull request by rayacers**
Add aggregate actions hook.

### CODE REVIEW

File: src/config/keys.ts [hunk 1]
```diff
@@ -191,6 +191,27 @@ export const buildActionsKey = (args: {
...
+}) => [
+  'aggregateActions',
+  args.itemId,
+  args.view,
+  args.requestedSampleSize,
+  args.type,
+  args.countGroupBy,
+  args.aggregateFunction,
+  args.aggregateMetric,
+  args.requestedSampleSize,
```
Comment (spaenleh) [inline src/config/keys.ts:212]: Maybe you could use the object directly as second argument. React-query allows to use objects as keys [react-query docs](https://tanstack.com/query/v3/docs/react/guides/query-keys)

File: src/hooks/action.test.ts [hunk 2]
```diff
@@ -94,4 +97,53 @@ describe('Action Hooks', () => {
       expect(queryClient.getQueryData(key)).toBeFalsy();
     });
   });
+
+  describe('useAggregateActions', () => {
+    const args = {
+      itemId,
+      view: 'builder',
```
Comment (spaenleh) [inline src/hooks/action.test.ts:104]: Use `Context.Builder` from sdk.

File: src/hooks/action.test.ts [hunk 3]
```diff
@@ -94,4 +97,53 @@ describe('Action Hooks', () => {
...
+      requestedSampleSize: 5,
+      type: 'update',
+      countGroupBy: ['createdDay', 'user'],
+      aggregateFunction: 'avg',
```
Comment (spaenleh) [inline src/hooks/action.test.ts:108]: Do you support a fixed number of functions ? Can you put them in an enum or a const object to make it easier to reference ? This also means that you can type the argument more strictly (avoid "string" type if only certain values are allowed).

File: src/hooks/action.ts [hunk 4]
```diff
@@ -4,34 +4,62 @@ import { UUID, convertJs } from '@graasp/sdk';
...
+    args: {
+      itemId: UUID;
+      view: string;
+      requestedSampleSize: number;
+      type: string;
+      countGroupBy: string[];
+      aggregateFunction: string;
+      aggregateMetric: string;
+      aggregateBy: string[];
+    },
```
Comment (spaenleh) [inline src/hooks/action.ts:46]: This type is defined in 3 places, maybe factor it out ?

**Review by spaenleh — COMMENTED** [review]
Thank you for the PR, here are some comments and suggestions. 
I can't approve as I am not following this feature very closely. Looking forward to the aggregated graphs in Analytics !

**Review by pyphilia — CHANGES_REQUESTED** [review]
I agree with everything Basile mentioned 👍

**Pull request closed without merging**